src/components/Registries/RegistriesCompanies/CompanyCRUD/index.js

```diff
@@ -1,38 +1,50 @@
-import React, {useEffect, useState} from 'react';
-import {useDispatch, useSelector} from "react-redux";
-import {blockchainSelectors, registriesSelectors} from "../../../../redux/selectors";
-import {newCompanyDataObject} from "../../../../utils/defaultData";
-import {buildRegistryForm} from "../../../../utils/registryFormBuilder";
-import {registriesActions} from "../../../../redux/actions";
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { registriesSelectors } from '../../../../redux/selectors';
+import { newCompanyDataObject } from '../../../../utils/defaultData';
+import { buildRegistryForm } from '../../../../utils/registryFormBuilder';
+import { registriesActions } from '../../../../redux/actions';
 
-const CompanyCRUD = () => {
-  const userWalletAddress = useSelector(blockchainSelectors.userWalletAddressSelector);
+function CompanyCRUD() {
+  const history = useHistory();
   const registryCRUDAction = useSelector(registriesSelectors.registryCRUDAction);
 
   const dispatch = useDispatch();
 
-  let formObject = {staticFields: [], dynamicFields: []}
-  let buttonMessage = ''
-  let companyId = false
-  let callback = (arg) => {console.log(arg)}
-  if (registryCRUDAction.registry == 'company') {
-    if(registryCRUDAction.action == 'create'){
-      formObject = newCompanyDataObject
-      buttonMessage = 'Submit Company Application'
-      callback = ((companyData) => {dispatch(registriesActions.requestCompanyRegistrationAction.call(companyData))})
+  let formObject = { staticFields: [], dynamicFields: [] };
+  let buttonMessage = '';
+  let companyId = false;
+  let callback = (arg) => {
+    // eslint-disable-next-line no-console
+    console.log(arg);
+  };
+  if (registryCRUDAction.registry === 'company') {
+    if (registryCRUDAction.action === 'create') {
+      formObject = newCompanyDataObject;
+      buttonMessage = 'Submit Company Application';
+      callback = ((companyData) => {
+        dispatch(
+          registriesActions.requestCompanyRegistrationAction.call({
+            companyData,
+            history,
+          }),
+        );
+      });
     }
-    if(registryCRUDAction.action == 'edit'){
-      formObject = registryCRUDAction.dataObject
-      buttonMessage = 'Submit Company Edit'
-      companyId = registryCRUDAction.dataObject.id
-      callback = ((companyData) =>{})
+    if (registryCRUDAction.action === 'edit') {
+      formObject = registryCRUDAction.dataObject;
+      buttonMessage = 'Submit Company Edit';
+      companyId = registryCRUDAction.dataObject.id;
+      callback = ((_companyData) => {});
```

Comment:
why the underscore ?

Reply:
We're trying to make new/modified code have no errors when linted with eslint. One of the lints is that unused variables should be prefixed with `_`. `_companyData` here is unused.